[SE-4175] Add CCX ID to generated filename prefixes

gabor-boros opened the pull request:
This pull requests adds a new feature to include CCX course IDs in the generated course filenames used for downloading reports, etc.

The suffix appended to the generated filename prefix is following the  `<separator>_ccx_<ccx id>` pattern, to ensure unique filename generation per ccx - previously all CCX courses were downloaded with the same name.

**Dependencies**: None

**Screenshots**: 
![Screenshot 2021-03-16 at 16 22 12](https://user-images.githubusercontent.com/19173947/111349994-f1c42980-8681-11eb-8fff-d28024899957.png)

**Sandbox URL**: Due to an `DeprecatedEdxPlatformImportError` exception the sandbox is not provisioned.

**Merge deadline**: None

**Testing instructions**:

1. Enable `ALLOW_COURSE_STAFF_GRADE_DOWNLOADS`, `ENABLE_GRADE_DOWNLOADS`, and `ENABLE_COURSE_FILENAME_CCX_SUFFIX` features.
2. Create a course
3. Enable CCX for the course in the advanced settings
4. Create a new CCX for the course
5. Generate and download grades

**Author notes and concerns**:

This change should be cherry-picked to Koa and Lilac as well.

**Reviewers**
- [ ] @mavidser 
- [ ] TBD

edx-webhook commented:
Thanks for the pull request, @gabor-boros! I've created [OSPR-5675](https://openedx.atlassian.net/browse/OSPR-5675) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

natabene commented:
@gabor-boros Thank you for your contribution. Please let me know once it is ready for our review.

gabor-boros commented:
@natabene I'll let you know as soon this is ready for a review.

mavidser commented on common/djangoapps/util/file.py:
```diff
@@ -89,16 +90,30 @@ def store_uploaded_file(
+    )
+
+    if enable_course_filename_ccx_suffix and getattr(course_id, 'ccx', None):
+        filename += separator.join([filename, 'ccx', course_id.ccx])
```
`filename = separator.join([filename, 'ccx', course_id.ccx])` - this is causing tests to fail.

mavidser reviewed (approved):
LGTM. This should be ready for edx's review once the tests pass :)

gabor-boros commented:
@natabene This is ready for edX review  🚀

nasthagiri commented:
@pdpinch Can you review and merge this PR related to CCX?

pdpinch commented:
jenkins run all

gabor-boros commented:
jenkins run all

gabor-boros commented:
FTR: Oh, there's an invalid syntax error:

```
18:04:13    File "/home/jenkins/workspace/edx-platform-quality-pipeline-pr/cms/envs/test.py", line 26, in <module>
18:04:13      from .common import *
18:04:13    File "/home/jenkins/workspace/edx-platform-quality-pipeline-pr/cms/envs/common.py", line 470
18:04:13      'ENABLE_HELP_LINK': True,
18:04:13      ^
18:04:13  SyntaxError: invalid syntax
```

gabor-boros commented:
@pdpinch FYI, I just fixed the failing tests 😇

natabene commented:
@pdpinch Could you have another look?

gabor-boros commented:
@natabene Could you please help me scheduling this PR for review? 😇

pdpinch commented:
Sorry I missed both your updates -- I'll take a look this week.

pdpinch commented:
Ok, I think I figured out the way to turn on CCX in devstack:

1. Set CUSTOM_COURSES_EDX in /edx/etc/lms.yml in the lms container. 
2. Restart the lms django process with `make lms-restart`
3. Set CUSTOM_COURSES_EDX in /edx/etc/studio.yml in the studio container
2. Restart the lms django process with `make studio-restart`
4. run migrations, just on lms: `make dev.lms.migrate`

gabor-boros commented on cms/envs/common.py:
```diff
@@ -443,6 +443,17 @@
+    # .. toggle_target_removal_date: None
+    # .. toggle_tickets: None
+    # .. toggle_warnings: Turning this feature ON will affect all generated filenames which are related to CCX courses.
+    'ENABLE_COURSE_FILENAME_CCX_SUFFIX': False,
```
Oh! I edited the wrong file! Thank you for pointing it out. I'll fix it in a bit.

gabor-boros replied:
@pdpinch fixed in https://github.com/edx/edx-platform/pull/27028/commits/96d5e9b4a1850d410a97ea8efbcdbf56c2d24966

gabor-boros commented:
@pdpinch And now I missed the ping. Sorry about that!

> I don't understand what you mean by this, in the PR description

I forgot to remove this sentence from the description. Removed it now!

> How are you enabling CCX for your devstack?
> Ok, I think I figured out the way

I removed that config on my side by now, but I can check if it still needed, but as I can see you found it, right? 😊

pdpinch reviewed (approved):
Looking good to me now. 👍

natabene commented:
@pdpinch Is this fit to merge? If so, let's merge.

pdpinch commented:
Sure thing! I wasn't sure if it needed another approval.

edx-webhook commented:
@gabor-boros 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-platform&entry.867055334=https://github.com/edx/edx-platform/pull/27028&entry.1484655318=https://github.com/gabor-boros&entry.752974735=2021-03-16+17:09&entry.1917517419=2021-05-25+14:25&entry.2133058324=Yes) so we can improve your experience in the future.
